plink_clump_hail.py

```diff
@@ -0,0 +1,366 @@
+    r'''
+    Adapted from ukb_common mwzj_hts_by_tree()
+    Uses read_clump_ht() instead of read_table()
+    '''
+    chunk_size = int(len(all_hts) ** 0.5) + 1
+    outer_hts = []
+    
+    if read_if_exists: print('\n\nWARNING: Intermediate tables will not be overwritten if they already exist\n\n')
+    
+    checkpoint_kwargs = {inner_mode: True if not read_if_exists else False,
+                         '_read_if_exists': read_if_exists} #
+    if repartition_final is not None:
+        intervals = ukb_common.get_n_even_intervals(repartition_final)
+        checkpoint_kwargs['_intervals'] = intervals
+    
+    if debug: print(f'Running chunk size {chunk_size}...')
+    for i in range(chunk_size):
+        if i * chunk_size >= len(all_hts): break
+        hts = all_hts[i * chunk_size:(i + 1) * chunk_size]
+        if debug: print(f'Going from {i * chunk_size} to {(i + 1) * chunk_size} ({len(hts)} HTs)...')
+        try:
+            if isinstance(hts[0], str):
+                def read_clump_ht(f):
+                    ht = hl.read_table(f)
+                    ht = ht.drop('idx')
+#                    ht = ht.select(F=ht.F,
+#                                   P=ht.P,
+#                                   TOTAL=ht.TOTAL,
+#                                   NSIG=ht.NSIG,
+#                                   S05=ht.S05,
+#                                   S01=ht.S01,
+#                                   S001=ht.S001,
+#                                   S0001=ht.S0001,
+#                                   SP2=ht.SP2)
+                    return ht
+                hts = list(map(read_clump_ht, hts))
+            ht = hl.Table.multi_way_zip_join(hts, 'row_field_name', 'global_field_name')
+        except:
+            if debug:
+                print(f'problem in range {i * chunk_size}-{i * chunk_size + chunk_size}')
+                _ = [ht.describe() for ht in hts]
+            raise
+        outer_hts.append(ht.checkpoint(f'{temp_dir}/temp_output_{i}.ht', **checkpoint_kwargs))
+    ht = hl.Table.multi_way_zip_join(outer_hts, 'row_field_name_outer', 'global_field_name_outer')
+    ht = ht.transmute(inner_row=hl.flatmap(lambda i:
+                                           hl.cond(hl.is_missing(ht.row_field_name_outer[i].row_field_name),
+                                                   hl.range(0, hl.len(ht.global_field_name_outer[i].global_field_name))
+                                                   .map(lambda _: hl.null(ht.row_field_name_outer[i].row_field_name.dtype.element_type)),
+                                                   ht.row_field_name_outer[i].row_field_name),
+                                           hl.range(hl.len(ht.global_field_name_outer))))
+    ht = ht.transmute_globals(inner_global=hl.flatmap(lambda x: x.global_field_name, ht.global_field_name_outer))
+    mt = ht._unlocalize_entries('inner_row', 'inner_global', globals_for_col_key)
+    return mt
+
+def resume_mwzj(temp_dir, globals_for_col_key):
```

Comment:
Heh this seems like a better approach than mine which was why `inner_mode` existed (though it still required reading the original HTs)